### PR TITLE
chore(lint): Disallow `var` in favor of `let` and `const`

### PR DESCRIPTION
**Pull request by pixelastic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "doctoc": "doctoc --maxlevel 3 README.md",
     "jsdoc:widget": "node ./scripts/doc/gen-widget-doc.js",
     "gh-pages": "./scripts/gh-pages.sh",
-    "lint": "eslint . --quiet --no-color && sass-lint --verbose",
+    "lint": "eslint ./components ./decorators ./lib ./widgets --quiet --no-color && sass-lint --verbose",
```
Comment (vvo) [inline package.json]: What is the failing directory? Tests should be linted also, not just the code.

Comment (vvo) [inline package.json]: If some file is failing and we want it to use var then we can just ignore it but our tests should be linted

Comment (pixelastic) [inline package.json]: I was aiming at excluding the webpack config and the `./scripts`.

Tests should be targeted as well, but you're right that something is not working as expected here. Will have another look.

Comment (vvo) [inline package.json]: Webpack config can definitely be linted also, along with the js scripts, would make sense to me

Comment (pixelastic) [inline package.json]: Fixed. I incorrectly thought that those files needed to go through babel and that would complexify the build process, but actually they are fine like this.

**Comment by vvo** [comment]
@algolia/javascript That's now official the instantsearch.js project is using `let` as the new `var`. For good or worse we do not know :D

**Comment by redox** [comment]
:+1: @pixelastic

**Comment by pixelastic** [comment]
Let's not forget about `const` as well, we can (and should) use it.

**Comment by redox** [comment]
> Let's not forget about const as well, we can (and should) use it.

https://github.com/algolia/instantsearch.js/blob/develop/widgets/search-box/search-box.js#L8 :)

**Comment by vvo** [comment]
That was the first const, constgrats

**Comment by olance** [comment]
may I ask what's the rationale behind this? ^^

**Comment by vvo** [comment]
I dont have an answer for this. Yet we can mix var and let but I am not sure of the value.

**Comment by Jerskouille** [comment]
@olance I think the main reason is hoisting.

**Comment by pixelastic** [comment]
`let` is not hoisted on the top of the function. Meaning you can now safely have `let` variables scoped inside `if` and `while` without any collision risk with a variable defined in a higher scope.

**Comment by olance** [comment]
OK, so it's a way to safeguard us from possible errors when using `var` in an inappropriate way?

**Comment by pixelastic** [comment]
Yes, it's just a `var` that works as someone discovering JavaScript would expect.

**Pull request merged**